Migrate the Xtend classes from the 'org.eclipse.xtext.ui' plugin to Java

nbhusare opened the pull request:
- [x] Migrate a subset of Xtend classes from the 'org.eclipse.xtext.ui' plugin to Java
- [x] Run the Maven/Gradle build locally

Signed-off-by: nbhusare neerajbhusare@gmail.com

cdietrich commented:
@nbhusare let us know when you have a workpackage done so that we can start reviewing

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	protected void handleReplacements(IEmfResourceChange change) {
+		if (change instanceof ITextDocumentChange) {
+			_handleReplacements((ITextDocumentChange) change);
+			return;
```
Unneeded

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			return;
+		} else if (change != null) {
+			_handleReplacements(change);
+			return;
```
Unneeded

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			_handleReplacements(change);
+			return;
+		} else {
+			throw new IllegalArgumentException("Unhandled parameter types: " + Arrays.<Object> asList(change).toString());
```
Does it need the `<Object>`?

nbhusare replied:
`<Object>` is not needed here.

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	}
+
+	protected ITextEditor findOpenEditor(IFile file) {
+		if ((this.workbench == null)) {
```
Unneeded parentheses

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+
+			@Override
+			protected ITextEditor run() throws Exception {
+				final FileEditorInput editorInput = new FileEditorInput(file);
```
Unneeded finals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			protected ITextEditor run() throws Exception {
+				final FileEditorInput editorInput = new FileEditorInput(file);
+				final IEditorReference[] editorReferences = workbench.getActiveWorkbenchWindow().getActivePage().getEditorReferences();
+				final Optional<ITextEditor> textEditorPart = Arrays.<IEditorReference> asList(editorReferences) //
```
Does it need the `<IEditorReference>`?

nbhusare replied:
not needed. I have removed it.

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+						.map(editorRef -> editorRef.getEditor(false)) //
+						.filter(ITextEditor.class::isInstance) //
+						.map(ITextEditor.class::cast) //
+						.filter(textEditor -> textEditor.getEditorInput() == editorInput) //
```
This should be `.equals` shouldn't it?

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	protected void handleUriChange(IEmfResourceChange change) {
+		if (change.getNewURI() != change.getOldURI()) {
+			if (change.getNewURI().lastSegment() == change.getOldURI().lastSegment()) {
+				final IFile oldFile = this.resourceUriConverter.toFile(change.getOldURI());
```
Unneeded finals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+
+	protected void handleUriChange(IEmfResourceChange change) {
+		if (change.getNewURI() != change.getOldURI()) {
+			if (change.getNewURI().lastSegment() == change.getOldURI().lastSegment()) {
```
This should be `.equals`

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	}
+
+	protected void handleUriChange(IEmfResourceChange change) {
+		if (change.getNewURI() != change.getOldURI()) {
```
`.equals`

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+				final MoveResourceChange ltkChange = new MoveResourceChange(oldFile, newContainer);
+				addChange(ltkChange);
+			} else {
+				if (change.getNewURI().trimSegments(1) == change.getOldURI().trimSegments(1)) {
```
.equals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	}
+
+	protected void _handleReplacements(ITextDocumentChange change) {
+		if (change.getReplacements().size() > 0) {
```
!isEmpty()

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+
+	protected void _handleReplacements(ITextDocumentChange change) {
+		if (change.getReplacements().size() > 0) {
+			final IFile file = resourceUriConverter.toFile(change.getOldURI());
```
Unneeded finals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			final MultiTextEdit textEdit = new MultiTextEdit();
+			textEdit.addChildren(textEdits.toArray(new TextEdit[textEdits.size()]));
+			final ITextEditor openEditor = findOpenEditor(file);
+			TextChange ltkChange = null;
```
Here's one where you can use final
`final TextChange ltkChange;`

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+	}
+
+	protected void _handleReplacements(IEmfResourceChange change) {
+		final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Unneeded finals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+				final String lastSegment = change.getOldURI().lastSegment();
+				final ReplaceFileContentChange ltkChange = new ReplaceFileContentChange(lastSegment, file, newContent);
+				addChange(ltkChange);
+			} catch (Throwable _e) {
```
catch (IOException e)

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+				final ReplaceFileContentChange ltkChange = new ReplaceFileContentChange(lastSegment, file, newContent);
+				addChange(ltkChange);
+			} catch (Throwable _e) {
+				throw Exceptions.sneakyThrow(_e);
```
Don't use sneakythrow

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+		this.name = name;
+	}
+
+	public ReplaceFileContentChange(final IFile file, final byte[] newContents) {
```
Remove the unneeded finals in this class

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+	public Change perform(final IProgressMonitor pm) throws CoreException {
+		final byte[] oldContents = this.getOldContents();
+		final ByteArrayInputStream newContentsIS = new ByteArrayInputStream(this.newContents);
+		return TryWithResource.<ReplaceFileContentChange> tryWith(newContentsIS, () -> {
```
No need for the generic

szarnekow replied:
Also no need for the `TryWithResource` utility.

nbhusare replied:
@szarnekow If I don't use the utility here, I'll be forced to duplicate the following piece of code (abstracted in the utility). 
```
try {                                      
	newContentsIS.close();                     
} catch (Throwable t) {                    
	LOG.error("Error closing stream", t);  
}                                          
```

szarnekow replied:
The throwable should be propagated as a CoreException rather than being only logged.

nbhusare replied:
@szarnekow Do you want me to import and use the utility statically, or you want me not to use the utility here?

szarnekow replied:
I want to get rid of `org.eclipse.xtext.ui.refactoring2.TryWithResource`

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+				file.setContents(newContentsIS, true, true, pm);
+				return new ReplaceFileContentChange(file, oldContents);
+			} catch (Throwable e) {
+				throw Exceptions.sneakyThrow(e);
```
No sneaky and a better catch would be nice

nbhusare replied:
Would you recommend throwing a RuntimeException here? Also in lines 91 and 95?

szarnekow replied:
It should be thrown as a CoreException as indicated by the method signature.

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+
+	protected byte[] getOldContents() {
+		final ByteArrayOutputStream oldContentsBAOS = new ByteArrayOutputStream();
+		return TryWithResource.<byte[]> tryWith(oldContentsBAOS, () -> {
```
No need for the generic

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+		return TryWithResource.<byte[]> tryWith(oldContentsBAOS, () -> {
+			try {
+				final InputStream oldContentsIS = file.getContents();
+				return TryWithResource.<byte[]> tryWith(oldContentsIS, () -> {
```
No generic

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+					try {
+						int readBytes = 0;
+						final byte[] buffer = new byte[4096];
+						while (((readBytes = oldContentsIS.read(buffer)) != (-1))) {
```
Unneeded parentheses

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+						}
+						return oldContentsBAOS.toByteArray();
+					} catch (Throwable _e) {
+						throw Exceptions.sneakyThrow(_e);
```
More sneaky

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,109 @@
+	}
+
+	@Override
+	@Pure
```
This is useless in java

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ResourceURIConverter.java:
```diff
@@ -27,30 +27,26 @@
+	@Inject(optional = true)
+	private IWorkspace workspace;
+
+	public URI toURI(final IResource file) {
```
Remove the unneeded finals

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/TryWithResource.java:
```diff
@@ -0,0 +1,48 @@
+		} finally {
+			try {
+				closeable.close();
+			} catch (final Throwable t) {
```
Remove the unneeded finals

nbhusare commented:
> @nbhusare let us know when you have a workpackage done so that we can start reviewing

@cdietrich I have completed the migration work. Please feel free to review.

cdietrich commented:
hi @nbhusare can you have a look the the conflicts.
any you also rebase and squash the commits into one

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hierarchy/AbstractCallHierarchyViewPart.java:
```diff
@@ -0,0 +1,180 @@
+
+	@Override
+	public void selectionChanged(SelectionChangedEvent event) {
+		if (event.getSelectionProvider() == callHierarchyViewer) {
```
Sure you don't mean .equals here?

szarnekow replied:
It was identity comparison in the original code.

tivervac commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hierarchy/HierarchyTreeContentProvider.java:
```diff
@@ -0,0 +1,75 @@
+	private DeferredTreeContentManager deferredTreeContentManager;
+
+	public HierarchyTreeContentProvider(ViewPart viewPart) {
+		super();
```
Not needed

nbhusare commented:
> hi @nbhusare can you have a look the the conflicts.
> any you also rebase and squash the commits into one

@cdietrich I am looking into the merge conflict. I tried rebasing the feature branch using `git rebase master` and it is up-to-date.

cdietrich commented:
@nbhusare thanks, please also squash the 12 commits into 1

cdietrich commented:
and you need to update your master:

> This branch is 36 commits behind eclipse:master.

nbhusare commented:
> and you need to update your master:
> 
> > This branch is 36 commits behind eclipse:master.

@cdietrich I verified my local master and it is up-to-date.

cdietrich commented:
yes but your master is behind upstream master
![Bildschirmfoto 2020-03-17 um 16 36 24](https://user-images.githubusercontent.com/238736/76873012-7e6cb980-686d-11ea-8b48-34c43c88c40a.png)

nbhusare commented:
@cdietrich I have fixed the issue. Please check now.

cdietrich commented:
hi, there are still 3 commits.

a rebase -i HEAD~3 and the squash the commits should do the trick

cdietrich commented:
also github still shows a merge conflict

cdietrich commented:
looks better now

cdietrich commented:
@szarnekow  what do you think?

szarnekow commented:
I'll go through this huge PR now. Please don't add more conversions to the existing change set.

szarnekow commented on org.eclipse.xtext.ui/deprecated/org/eclipse/xtext/ui/editor/copyqualifiedname/CopyQualifiedNameService.java:
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2018 itemis AG (http://www.itemis.eu) and others.
+ * Copyright (c) 2013, 2018, 2020 itemis AG (http://www.itemis.eu) and others.
```
Cosmetics: Usually inly two years are listed in the copyright header, e.g `2013, 2020`

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/bracketmatching/BracePairMatcher.java:
```diff
@@ -0,0 +1,72 @@
+import org.eclipse.jface.text.source.DefaultCharacterPairMatcher;
+import org.eclipse.jface.text.source.ICharacterPairMatcher;
+import org.eclipse.xtext.ide.editor.bracketmatching.IBracePairProvider;
+import org.eclipse.xtext.xbase.lib.Conversions;
```
I don't really like *xbase imports in converted files.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/UiToIdeContentProposalProvider.java:
```diff
@@ -0,0 +1,147 @@
+	@Override
+	public void createProposals(org.eclipse.xtext.ui.editor.contentassist.ContentAssistContext context,
+			ICompletionProposalAcceptor acceptor) {
+		ArrayList<Pair<ContentAssistEntry, Integer>> entries = new ArrayList<Pair<ContentAssistEntry, Integer>>();
```
Could use SimpleEntry rather than xbase's Pair.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hierarchy/AbstractHierarchyViewPart.java:
```diff
@@ -0,0 +1,92 @@
+
+	protected <T extends IHierarchyBuilder> T getBuilder(Class<T> clazz) {
+		if (clazz.isInstance(builder)) {
+			return ((T) builder);
```
I prefer `clazz.cast(..)` when dealing with type parameters.

szarnekow replied:
There must be warnings in this code here without that, aren't there?

nbhusare replied:
You are right. There is a warning. Will make use of `clazz.cast(...)`. Tx

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hierarchy/DeferredHierarchyBuilder.java:
```diff
@@ -0,0 +1,155 @@
+		@Override
+		public <T extends Object> T getAdapter(Class<T> adapterType) {
+			if (adapterType.isInstance(this)) {
+				return ((T) this);
```
`adapterType.cast(..)`

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hierarchy/HierarchyTreeContentProvider.java:
```diff
@@ -0,0 +1,75 @@
+	@Override
+	public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
+		if (viewer instanceof AbstractTreeViewer) {
+			deferredTreeContentManager = new DeferredTreeContentManager(((AbstractTreeViewer) viewer), viewPart.getSite());
```
Cosmetics: unnecessary parentheses in the argument list.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/navigation/NavigationService.java:
```diff
@@ -0,0 +1,113 @@
+	public void open(OpenEvent openEvent, boolean select) {
+		final ISelection selection = openEvent.getSelection();
+		if (selection instanceof IStructuredSelection) {
+			Streams.stream(((IStructuredSelection) selection).iterator()) //
```
A little stream-api-happy here ... a simple for-loop would be easier to read / debug. Not a request to rewrite this now, though.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/navigation/NavigationService.java:
```diff
@@ -0,0 +1,113 @@
+			final Object navigatableElement = navigatable.getNavigationElement();
+			if (navigatableElement instanceof IReferenceDescription) {
+				uriEditorOpener.open( //
+						((IReferenceDescription) navigatableElement).getSourceEObjectUri(), //
```
A local variable of type IReferenceDescription would be helpful here.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/quickfix/WorkbenchMarkerResolutionAdapter.java:
```diff
@@ -0,0 +1,184 @@
+	private IMarker primaryMarker;
+
+	@Override
+	public IMarker[] findOtherMarkers(final IMarker[] markers) {
```
A couple of unnecessary final parameters in this class.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/quickfix/WorkbenchMarkerResolutionAdapter.java:
```diff
@@ -0,0 +1,184 @@
+	@Override
+	public IMarker[] findOtherMarkers(final IMarker[] markers) {
+		return Arrays.asList(markers).stream()
+				.filter(marker -> marker != primaryMarker && issueUtil.getCode(primaryMarker) == issueUtil.getCode(marker))
```
Should be `.equals()` for the getCode comparison.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/quickfix/WorkbenchMarkerResolutionAdapter.java:
```diff
@@ -0,0 +1,184 @@
+		final SubMonitor monitor = SubMonitor.convert(progressMonitor);
+		monitor.beginTask("Applying resolutions", markersByProject.size());
+
+		markersByProject.entrySet().forEach(e -> {
```
I think map does have a forEach, too.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/quickfix/WorkbenchMarkerResolutionAdapter.java:
```diff
@@ -0,0 +1,184 @@
+	private boolean isSameResolution(final IssueResolution issueResolution, final IssueResolution other) {
+		return issueResolution != null //
+				&& other != null //
+				&& issueResolution.getDescription() == other.getDescription() //
```
equals vs ==

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/reconciler/CancelIndicatorBasedProgressMonitor.java:
```diff
@@ -0,0 +1,69 @@
+ */
+public class CancelIndicatorBasedProgressMonitor implements IProgressMonitor {
+
+	private final IProgressMonitor doDelegate = new NullProgressMonitor();
```
doDelegate is a weird fieldname

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/EclipseBasedShouldGenerate.java:
```diff
@@ -0,0 +1,58 @@
+
+	@Override
+	public boolean shouldGenerate(Resource resource, CancelIndicator cancelIndicator) {
+		final URI uri = resource.getURI();
```
Unnecessary final locals

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/EclipseBasedShouldGenerate.java:
```diff
@@ -0,0 +1,58 @@
+					try {
+						return member.findMaxProblemSeverity(null, true, IResource.DEPTH_INFINITE) != IMarker.SEVERITY_ERROR;
+					} catch (CoreException e) {
+						throw Exceptions.sneakyThrow(e);
```
I don't think we really want to throw. Better log here and return false.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/EclipseBasedShouldGenerate.java:
```diff
@@ -0,0 +1,58 @@
+			final ProjectConfigAdapter projectConfigAdapter = ProjectConfigAdapter.findInEmfObject(resource.getResourceSet());
+			if (projectConfigAdapter != null) {
+				final IProjectConfig projectConfig = projectConfigAdapter.getProjectConfig();
+				if (projectConfig != null && member.getProject().getName() == projectConfig.getName()) {
```
equals vs ==

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+
+	protected void _handleReplacements(IEmfResourceChange change) {
+		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+		tryWith(outputStream, () -> {
```
Please use the native try with resource instead

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			try {
+				IFile file = resourceUriConverter.toFile(change.getOldURI());
+				if (!canWrite(file)) {
+					issues.add(RefactoringIssueAcceptor.Severity.ERROR, "Affected file '" + file.getFullPath() + "' is read-only");
```
missing return? Even though the Xtend original simply ignores that diagnostic, too.

nbhusare replied:
Are you suggesting to **add** a return statement on line 128?

szarnekow replied:
Indeed. If the file is readonly, it’s probably bad to attempt a modification

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java:
```diff
@@ -0,0 +1,245 @@
+			protected ITextEditor run() throws Exception {
+				FileEditorInput editorInput = new FileEditorInput(file);
+				IEditorReference[] editorReferences = workbench.getActiveWorkbenchWindow().getActivePage().getEditorReferences();
+				Optional<ITextEditor> textEditorPart = Arrays.asList(editorReferences) //
```
A general remark: Too many streams for my liking.
Eclipse stream debugging support is not ready for prime time yet and thus I tend to rewrite these to normal for loops more often than I enjoy having the stream API

szarnekow replied:
@cdietrich Wasn't there a util to find an open editor for an editor input?

nbhusare replied:
@szarnekow `workbench.getActiveWorkbenchWindow().getActivePage().findEditor` can be used here. The logic can be simplified to

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,111 @@
+	}
+
+	public ReplaceFileContentChange(IFile file, byte[] newContents) {
+		super();
```
unnecessary super()

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java:
```diff
@@ -0,0 +1,111 @@
+	}
+
+	protected byte[] getOldContents() {
+		byte[] oldContents = new byte[0];
```
Use com.google.common.io.ByteStreams.toByteArray(InputStream) instead.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/TryWithResource.java:
```diff
@@ -0,0 +1,48 @@
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+public class TryWithResource {
```
Please mark as deprecated

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/XtextCopyResourceParticipant.java:
```diff
@@ -0,0 +1,85 @@
+			return processor.getIssues().getRefactoringStatus();
+		} catch (OperationCanceledException e) {
+			throw e;
+		} catch (CoreException e) {
```
I'm sure this should better become part of the result, e.g. something like RefactoringStatus .FATAL should be put in use here.

nbhusare replied:
I can do `return RefactoringStatus.createFatalErrorStatus("Error copying resource participant");` . @szarnekow @cdietrich Is the ok to have the above error message?

szarnekow replied:
You could use RefactoringStatus.create(coreException.getStatus)

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/XtextCopyResourceParticipant.java:
```diff
@@ -0,0 +1,85 @@
+				Object destination = ((CopyArguments) arguments).getDestination();
+				if (destination instanceof IFolder || destination instanceof IProject) {
+					IFile destinationFile = null;
+					if (destination instanceof IFolder) {
```
can be simplified to IContainer I assume

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/XtextMoveResourceParticipant.java:
```diff
@@ -0,0 +1,90 @@
+		} catch (OperationCanceledException e) {
+			throw e;
+		} catch (CoreException e) { 
+			throw new RuntimeException(e);
```
See remark in the other file

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/rename/DefaultLinkedPositionGroupCalculator2.java:
```diff
@@ -0,0 +1,194 @@
+			throw new OperationCanceledException();
+		}
+
+		IRenameStrategy2 renameStrategy = globalServiceProvider.<IRenameStrategy2> findService(targetElement, IRenameStrategy2.class);
```
Unnecessary type arguments in this class

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/rename/DefaultLinkedPositionGroupCalculator2.java:
```diff
@@ -0,0 +1,194 @@
+				String textToReplace = document.get(replaceEdit.getOffset(), replaceEdit.getLength());
+				int indexOf = textToReplace.indexOf(originalName);
+				if (indexOf != -1) {
+					int calculatedOffset = (replaceEdit.getOffset() + indexOf);
```
unnecessary parentheses

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/rename/ISimpleNameProvider.java:
```diff
@@ -0,0 +1,70 @@
+		}
+
+		protected Optional<EAttribute> getNameEAttribute(EObject target) {
+			return target.eClass() //
```
I don't think that we want to use Optional here. Also this is a change to the return type.

nbhusare replied:
Agree. I'll make the required change.

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/rename/RenameElementProcessor2.java:
```diff
@@ -0,0 +1,201 @@
+	@Inject
+	private ChangeConverter.Factory changeConverterFactory;
+
+	@Accessors({ AccessorType.PUBLIC_GETTER, AccessorType.PUBLIC_SETTER })
```
Unnecessary annotations

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/rename/RenameElementProcessor2.java:
```diff
@@ -0,0 +1,201 @@
+		RenameContext renameContext = new RenameContext(Lists.newArrayList(change), resourceSet, changeSerializer, status);
+		renameStrategy.applyRename(renameContext);
+
+		String name = new StringBuilder("Rename ") //
```
String + String is simple for this use case

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/tasks/TaskMarkerContributor.java:
```diff
@@ -0,0 +1,73 @@
+		try {
+			file.deleteMarkers(TaskMarkerTypeProvider.XTEXT_TASK_TYPE, true, IResource.DEPTH_ZERO);
+		} catch (Throwable _e) {
+			throw Exceptions.sneakyThrow(_e);
```
wrap in runtime exception

szarnekow commented on org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/workspace/EclipseProjectConfig.java:
```diff
@@ -0,0 +1,103 @@
+	private final EclipseProjectConfigProvider projectConfigProvider;
+
+	public EclipseProjectConfig(IProject project, EclipseProjectConfigProvider projectConfigProvider) {
+		super();
```
Unnecessary super

nbhusare commented:
@miklossy @szarnekow @tivervac I was facing issues while correcting the commit history in this branch. After discussion the issue with @cdietrich , I have raised another PR (https://github.com/eclipse/xtext-eclipse/pull/1374) where I have ported all the changes from this branch, and also addressed all your review comments. Kindly have a look. Sorry for the inconvenience.
